src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,241 @@
+            Ole32.CreateBindCtx(0, out bindContext);
+
+            do
+            {
+                monikers[0] = null;
+
+                var monikersFetched = 0u;
+                var result = enumMoniker.Next(1, monikers, out monikersFetched);
+
+                if (result != 0)
+                {
+                    return null;
+                }
+
+                var moniker = monikers[0];
+                var fullDisplayName = null as string;
+
+                moniker.GetDisplayName(bindContext, null, out fullDisplayName);
+
+                var displayNameProcessId = 0;
+
+                if (!int.TryParse(fullDisplayName.Split(':').Last(), out displayNameProcessId))
+                {
+                    continue;
+                }
+
+                var displayName = fullDisplayName.Substring(0, (fullDisplayName.Length - (displayNameProcessId.ToString().Length + 1)));
+                var fullProgId = vsProgId.StartsWith("!") ? vsProgId : $"!{vsProgId}";
+
+                if (displayName.Equals(fullProgId, StringComparison.OrdinalIgnoreCase) &&
+                    (displayNameProcessId == process.Id))
+                {
+                    runningObjectTable.GetObject(moniker, out dte);
+                }
+            }
+            while (dte == null);
+
+            return (DTE)(dte);
+        }
+
+        public static async Task<string> LocateFile(string fileName, string downloadUrl = "")
+        {
+            if (File.Exists(fileName))
+            {
+                return Path.GetFullPath(fileName);
+            }
+
+            var filePath = Path.Combine(Environment.CurrentDirectory, fileName);
+
+            if (File.Exists(filePath))
+            {
+                return filePath;
+            }
+
+            filePath = Path.Combine(typeof(IntegrationHelper).Assembly.Location, fileName);
+
+            if (File.Exists(filePath))
+            {
+                return filePath;
+            }
+
+            if (!string.IsNullOrWhiteSpace(downloadUrl))
+            {
+                await DownloadFileAsync(downloadUrl, fileName).ConfigureAwait(continueOnCapturedContext: false);
+            }
+
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException($"Unable to locate the specified file. (FileName: '{fileName}')", fileName);
+            }
+            
+            return filePath;
+        }
+
+        public static object GetRegistryKeyValue(RegistryKey baseKey, string subKeyName, string valueName)
+        {
+            using (var registryKey = baseKey.OpenSubKey(subKeyName))
+            {
+                if (registryKey == null)
+                {
+                    throw new Exception($"The specified registry key could not be found. Registry Key: '{registryKey}'");
+                }
+
+                return registryKey.GetValue(valueName);
+            }
+        }
+
+        public static bool TryDeleteDirectoryRecursively(string path)
```

Comment:
Because there is other cleanup that needs to be done to ensure the next test isn't blocked.